.circleci/config.yml

```
@@ -1,32 +1,74 @@
-  code-style:
-    docker:
-      - image: circleci/node:latest
+  build:
+    <<: *defaults
     steps:
       - checkout
+      - restore_cache:
+          keys:
+            - cache-{{ .Branch }}-{{ checksum "package.json" }}
       - run:
           name: Install NPM dependencies
-          command: npm install
+          command: npm ci
       - run:
-          name: Check Code Style
+          name: Verify Cypress
+          command: npm run cypress-verify
+      - save_cache:
+          key: cache-{{ .Branch }}-{{ checksum "package.json" }}
+          paths:
+            - ~/.npm
+            - ~/.cache
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - app
+            - .cache/Cypress
+
+  code-style:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Check code style
           command: npm run style
 
-  test:
-    docker:
-      - image: circleci/node:latest
+  unit-tests:
+    <<: *defaults
     steps:
-      - checkout
+      - attach_workspace:
+          at: ~/
       - run:
-          name: Install NPM Dependencies
-          command: npm install
+          name: Perform unit tests and send coverage data
+          command: npm run jest-tests-with-coverage
+
+  browser-integration-tests:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/
       - run:
-          name: Perform Unit Tests and Send Coverage Data
-          command: npm run test-with-coverage
+          name: Run browser integration tests and record data
+          command: npx start-server-and-test develop http://localhost:8000 cypress-run-record
 
 workflows:
   version: 2
   run-tests:
     jobs:
-      - code-style
-      - test
+      - build
+      - code-style:
+          requires:
+            - build
+      - unit-tests:
+          requires:
+            - build
+            - code-style
+      - browser-integration-tests:
+          requires:
+            - build
```

Now this job won't run unless `build` and `unit-tests` have succeeded, this makes failing faster because we don't have to wait for all to fail to get the status back on the webhook not to mention waste compute power.